docs/content/python/getting-started/application/model.md

```diff
@@ -0,0 +1,214 @@
+this guide, we're going to show how to implement the most common
+authorization model, role-based access control, with Oso. See [How to:
+modeling](TODO) **TODO** to see how to implement other authorization
+models in Oso.
+
+To start with, you need to define resource blocks in your policy. Let's
+say we have an application like GitHub that includes a `Repository`
+resource. Define the `Repository` resource in `main.polar`:
+
+```polar
+actor User {}
+
+resource Repository {
+    permissions = ["read", "push"];
+    roles = ["contributor", "maintainer"];
+
+    # A user has the "read" permission if they have the
+    # "contributor" role.
+    "read" if "contributor";
+
+    # A user has the "push" permission if they have the
+    # "maintainer" role.
+    "push" if "maintainer";
+}
+```
+
+This policy declares `"read"` and `"push"` as permissions for the
+`Repository` resource and assigns them to roles. We also tell Oso
+that our `User` class is an `actor` with the `actor User {}`
+declaration.
+
+The `"read" if "contributor";` statement is an example of a *shorthand rule.*
+Add an `"admin"` role and give it its own permissions by adding some
+more shorthand rules:
+
+```polar
+resource Repository {
+    permissions = ["read", "push", "delete"];
+    roles = ["contributor", "maintainer", "admin"];
+
+    "read" if "contributor";
+    "push" if "maintainer";
+
+	# A user has the "delete" permission if they have the
+	# "admin" role.
+	"delete" if "admin";
+
+	# A user has the "maintainer" role if they have
+	# the "admin" role.
+    "maintainer" if "admin";
+
+	# A user has the "contributor" role if they have
+	# the "maintainer" role.
+    "contributor" if "maintainer";
+}
+```
+
+The last rules we added are between two roles: A user has the
+`"maintainer"` role and all permissions associated with it if they have the
+`"admin"` role, and the same for `"contributor"` and `"maintainer"`.
+
+### Giving your users roles
+
+Now that we've finished our policy, we must associate users with roles
+in our application. Oso doesn't manage authorization data. The data
+stays in your application's existing data store.
+
+All the data we've defined so far in the policy is static: it isn't
+changed by end users of the application. The development team
+modifies permission associations with roles, and the list of roles for
+each resource by updating the policy. But, some parts of this policy
+must be dynamic: the association of users with a role.
+
+Write a `has_role` rule to tell Oso whether your users have a particular
+role:
+
+```polar
+has_role(user: User, role_name: String, repo: Repository) if
+  role in actor.roles and
+  role_name = role.name and
+  repository = role.repository;
+```
+
+{{% minicallout %}}
+This is an example of a full Polar rule. We'll go more into writing
+rules in the [Write Polar rules](write-rules).
+{{% /minicallout %}}
+
+The `has_role` rule uses the user object passed into Oso by your
```

Comment:
This was tough to explain because I don't really want to fully introduce rules, and I think there are a lot of concepts involved to completely understand what's happening here.

The explanation here is very literal: just what's happening during evaluation because I am assuming users will need to read the write rules guide to fully understand how to write their own `has_role` rules. It is more correct to say it expresses the logical conditions, but that doesn't tell someone how it executes.

I also want to emphasize that these objects are being passed in from the application so that someone takes away that key concept.

Need to think about this a bit more.